app/models/event.rb

```diff
@@ -1,6 +1,8 @@
 # frozen_string_literal: true
 class Event < ApplicationRecord
-  KINDS = %w(user_present_on_map conversation_started_on_map topic_added_to_map synapse_added_to_map).freeze
+  KINDS = %w(user_present_on_map conversation_started_on_map 
+               topic_added_to_map topic_moved_on_map topic_removed_from_map
+               synapse_added_to_map synapse_removed_from_map).freeze
```

Comment:
By capturing this particular set of events:
```
topic_added_to_map
topic_moved_on_map
topic_removed_from_map
synapse_added_to_map
synapse_removed_from_map
```
and storing the x and y values of the topic at the time of creation, and then every time it moves, we have everything we need to reconstruct the current state of the map, and watch its evolution over time.